src/frontend/src/table.rs

```diff
@@ -348,6 +352,26 @@ impl DistTable {
         };
         Ok(partition_rule)
     }
+
+    /// Define a `alter_by_expr` instead of impl [`Table::alter`] to avoid redundant conversion between  
+    /// [`table::requests::AlterTableRequest`] and [`AlterExpr`].
+    pub(crate) async fn alter_by_expr(&self, expr: AlterExpr) -> Result<()> {
```

Comment:
I think we should implement DistTable::alter and use it. AlterTable -> AlterTableRequest -> AlterExpr for DistTable is fine, doesn't have to "must use" the `alter_expr` in Frontend instance [here](https://github.com/GreptimeTeam/greptimedb/pull/572/files#diff-7b73fa585fc3ed84f34902a8126a169043dcf771cb9a2961d3795bab34db221eR272)

Reply:
When it comes to alter table on insertion, it would be `AlterExpr`->`AlterTableRequest`->`AlterExpr`